File: algorithms/EuclideanAlgorithm/javascript/euclideanAlgorithm.js
function euclideanAlg(num1, num2) {
    let numA = num1;
    let numB = num2;
  
    while (true) {
      const mod = numA % numB;
  
      if (mod === 0) {
        break;
      }
  
      if (mod > 0) {
        numA = numB;
        numB = mod;
      }
    }
  
    return numB;
  }

File: algorithms/EuclideanAlgorithm/cpp/euclideanAlgorithm.cpp
//
// Created by Jonatan Witoszek on 21.10.2017.
//

#include <iostream>

int euclidean(int num1, int num2) {
    int a = num1;
    int b = num2;

    if (a && b == 0) {
        return 0; // returns 0 when both values are 0 too
    }

    int rem = 0;

    while(b!=0) {
        rem = b;
        b = a % b;
        a = rem;
    }

    return rem;

}

int main() {
    std::cout << "Specify two numbers to calculate greatest common divisor: ";

    int a, b;
    std::cin >> a >> b;

    std::cout << "gcd(" << a << ", " << b << ") = " << euclidean(a,b);

    std::cin.get();
    std::cin.get();
    return 0;
}
